openapi-cli/src/main/java/io/ballerina/openapi/cmd/OASContractGenerator.java

```diff
@@ -102,24 +99,34 @@ public void generateOAS3DefinitionsAllService(Path servicePath, Path outPath, St
         Optional<Path> path = project.documentPath(docId);
         Path inputPath = path.orElse(null);
 
-        syntaxTree = doc.syntaxTree();
+        SyntaxTree syntaxTree = doc.syntaxTree();
         PackageCompilation compilation = project.currentPackage().getCompilation();
         boolean hasCompilationErrors = compilation.diagnosticResult()
                 .diagnostics().stream()
                 .anyMatch(d -> DiagnosticSeverity.ERROR.equals(d.diagnosticInfo().severity()));
         if (hasCompilationErrors || hasErrorsFromCodeGenAndModify) {
-            // if there are any compilation errors, do not proceed
+            // if there are any compilation errors, do not proceed and those diagnostic will display to user
+            outStream.println("OpenAPI contract generation failed due to Ballerina code has compilation errors. :");
+            compilation.diagnosticResult().diagnostics().forEach(diagnostic -> {
+                if (diagnostic.diagnosticInfo().severity().equals(DiagnosticSeverity.ERROR)) {
```

Comment:
i remember there was an api for this like `diagnostic.diagnosticInfo().hasErrors()` or something like that